Add Annual Security Reports

jacobdjwilson opened the pull request:
<!-- Congrats on creating an Awesome list! 🎉 -->

<!-- Please fill in the below placeholders -->

[**[Insert URL to the list here]**]
(https://github.com/jacobdjwilson/awesome-annual-security-reports)

**[Explain what this list is about and why it should be included here]**
A curated list of annual cyber security reports.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- [X] Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it. **Instead use [#2242](https://github.com/sindresorhus/awesome/issues/2242) for incubation visibility**.
- [X] **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- [X] **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
- [#2503](https://github.com/sindresorhus/awesome/pull/2503#issuecomment-1483041705),[#2498](https://github.com/sindresorhus/awesome/pull/2498#issuecomment-1483075027)
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions. Comments pointing out lint violation are allowed, but does **not** count as a review.
- [X] You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/main/create-list.md).
- [X] This pull request has a title in the format `Add Name of List`. It should not contain the word `Awesome`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `add Swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- [X] Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- [X] Your entry should be added at the bottom of the appropriate category.
  **Security**
  This PR should add the entry to the readme.md
- [X] The title of your entry should be title-cased and the URL to your list should end in `#readme`.
	- Example: `- [Software Architecture](https://github.com/simskij/awesome-software-architecture#readme) - The discipline of designing and building software.`
- [X] The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- [X] **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- [X] Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
  **Can someone help me with this?**
- [X] The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- [X] **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- [X] It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- [X] The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- [X] The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- [X] Non-generated Markdown file in a GitHub repo.
- [X] The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- [X] Not a duplicate. Please search for existing submissions.
- [X] Only has awesome items. Awesome lists are curations of the best, not everything.
- [X] Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
  **How long does the group suggest aging out reports?**
- [X] Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- [X] Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- [X] Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/main/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- [X] Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
	- Must not feature `Contributing` or `Footnotes` sections.
- [X] Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=main&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name, text, or a `Licence` section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- [X] Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/main/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
	- It can optionally be linked from the readme in a dedicated section titled `Contributing`, positioned at the top or bottom of the main content.
	- The section should not appear in the Table of Contents.
- [X] All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a `Footnotes` section at the bottom of the readme. The section should not be present in the Table of Contents.
- [X] Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- [X] Does not use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- [X] Does not include a CI (e.g. GitHub Actions) badge.<br>You can still use a CI for linting, but the badge has no value in the readme.
- [X] Does not include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.
  **I put other non-awesome inspirational people, does that count?**

**Go to the top and read it again.**

sindresorhus commented:
Thanks for making an Awesome list! 🙌

It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

jacobdjwilson commented:
> Thanks for making an Awesome list! 🙌
> 
> It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

To my knowledge the only requirement not fulfilled is linting. Is there someone that can help lint this? I don't see a clear set of directions here. Also this awesome list has existed for well over a year.

jacobdjwilson reviewed (commented):
Add Annual Security Reports

roj1512 reviewed (commented):
Hi, @jacobdjwilson. Here are some violations of the guidelines I’ve found while looking into your submission:

- Your pull request’s title is not in the suggested format.
- Your table of contents includes the “Contributing” section.
- awesome-lint fails: Your list items are neither consistent, nor following the guidelines.

jacobdjwilson commented:
Thank you @roj1512 !! I think i've fixed the output of the lint run and your feedback on the contribution and title

roj1512 commented:
@jacobdjwilson The title of this pull request is still not in the suggested format.

jacobdjwilson commented:
Gotcha, I've modified the PR title as well.

The one thing I can't seem to fix in awesome-lint is `remark-lint:double-link`
Some items in the list will link to the same vendor and may cause double links in this case.

roj1512 commented:
@jacobdjwilson Above the line that is unreasonably failing, you can put this comment:

```html
<!-- lint ignore double-link -->
```

jacobdjwilson commented:
Okay perfect, I put the ignore rule on line 1. We should be good to go now! I appreciate all the help!

sindresorhus commented:
> The description starts with an uppercase character and ends with a period.

sindresorhus commented:
For each item in the `Analysis` and `Survey` section, it would be useful with a one-sentence summary for each item. You can use ChatGPT for this.

I also think the report name column should be first.

jacobdjwilson commented:
@sindresorhus Is there any way to get this to pass the final failing lint check?

jacobdjwilson reviewed (commented):
Please add Annual Security Reports

sindresorhus commented:
This review feedback is not addressed. You are also not following all the guidelines.

jacobdjwilson commented:
> This review feedback is not addressed. You are also not following all the guidelines.

It's been over 6 months since this PR was opened and unfortunately I am unable to get the context to pass the linting checks for guidelines.

Respectfully, the feedback in your last post was not helpful in moving this PR forward. I disagree that the report title should be listed in the first column, if you review the dataset you'll see that a vast majority of reports have the name "Annual Threat Report" and the Provider column not only is more unique, but also allows users to quickly find multiple reports by relevant suppliers to their security program. Additionally, the use of generic ChatGPT sentence summaries degrades the usefulness of the list in my opinion. The most useful context in that sentence summary is already contained within the Data Type column.

I've put alot of thought into the structure and usability of this particular list, as I see that you have put alot of thought into the Awesome project as a whole. It would be great to get this PR merged.

jacobdjwilson commented:
@sindresorhus Okay I've reformatted all the data in list format and addressed your comments. Could you please review, thanks!

Bongdautay commented:
quaroibn1@gmail.com

jacobdjwilson commented:
@sindresorhus Could you please review the latest main branch of awesome-annual-security-reports? Thanks

jacobdjwilson commented:
@rojvv I have changed the format of the main branch and moved the old format to a legacy branch. You were super helpful in previous comments helping with lint checks and I appreciate that, could you help with re-running the checks? Thank you!

rojvv commented:
I helped because I had a PR of mine open back then, to fulfill its requirements. It is closed now and for the same reason I don't have a reason to help anymore. Sorry about that.

emmanuelgautier commented:
There are several linting issues noted [here](https://github.com/sindresorhus/awesome/actions/runs/11556633315/job/32164771641?pr=2850#step:3:106). You can also run the lint command locally to check.

For example, the error ``List item description must start with valid casing`` means that your description should start with an uppercase letter, which isn't the case in your README for those specific lines.

jacobdjwilson reviewed (commented):
Add Annual Security Reports

jacobdjwilson reviewed (commented):
Add Annual Security Reports

jacobdjwilson commented:
@sindresorhus Can this PR be merged please? Thanks

sindresorhus commented:
> Mandiant - MTrends Special Report (2024) - The MTrends Special Report offers insights into advanced persistent threats, emerging attack techniques, and strategies for improving organizational cyber defense.

The description should not start with the title. Applies in multiple places.

sindresorhus commented on readme.md:
```diff
@@ -632,6 +632,7 @@
 
 ## Security
 
+- [Annual Security Reports](https://github.com/jacobdjwilson/awesome-annual-security-reports#readme) - Annual cybersecurity analysis and industry survey reports
```
Should be added at the bottom.

sindresorhus replied:
> Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.**

jacobdjwilson replied:
I have moved the entry to the bottom of the list and added a meaningful description

jacobdjwilson commented:
> > Mandiant - MTrends Special Report (2024) - The MTrends Special Report offers insights into advanced persistent threats, emerging attack techniques, and strategies for improving organizational cyber defense.
> 
> The description should not start with the title. Applies in multiple places.

I have removed all mention of report titles in the description and any references to "the report" or other circular references.

sindresorhus commented:
Tweet: https://x.com/awesome__re/status/1859890268746756201
